### ISSUE TITLE
[labs/react] Import of wrapped component in Node/Next.js errors

### ISSUE THREAD

**Issue by augustjk** [lit/lit#3397]
### Which package(s) are affected?

React (@lit-labs/react)

### Description

https://github.com/lit/lit/issues/3079 allowed importing Lit components in Node but components wrapped with `@lit-labs/react` breaks with following error

```
error - ReferenceError: HTMLElement is not defined
    at r.render (file:///home/projects/nextjs-mgpt7h/node_modules/@lit-labs/react/create-component.js:19:1017)
```

This happens when trying to import the wrapped component in Next.js.

### Reproduction

https://stackblitz.com/edit/nextjs-mgpt7h?file=pages%2Findex.js

### Workaround

For Next.js, use `next/dynamic` to lazy load. See commented out code in repro above.

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

1.0.9

### Browser/OS/Node environment

N/A

**Comment by taylor-vann** [comment]
My bad for not paying more attention to this bug. I'd like to provide feedback and alternatives before I'd like to talk alternatives before we couple `node` to `@lit-labs/react`

**Comment by augustjk** [comment]
The immediate error during React server rendering is fixed by #3128 (Thanks @taylor-vann !)

There's some janky behavior on initial render though. Detailed in #3418

**Comment by taylor-vann** [comment]
That's wonderful to hear! Thanks for working on this one @augustjk, really appreciate it
